server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -343,4 +344,13 @@ export class AssetRepository implements IAssetRepository {
     });
     return new CheckExistingAssetsResponseDto(existingAssets.map((a) => a.deviceAssetId));
   }
+
+  async countByIdAndUser(assetId: string, userId: string): Promise<number> {
+    return await this.assetRepository.count({
```

Comment:
Are we using `count` instead of `findOne` for performance purposes?

Reply:
Yep, that was my thought. This runs for every thumbnail so I wanted the amount of data that has to be mapped to objects as small as possible.

Reply:
Can you grab a quick benchmark for the runtime of the `checkAssetsAccess` method when we have 10 vs 100 vs 200 assets in the database?

Reply:
I'll do some measurements later.

Reply:
![image](https://user-images.githubusercontent.com/6898797/203408729-42dc3c5c-b9e1-40bf-9794-d9a58643d2c4.png)

I measured the time the check takes. There seems to be no difference between 100, 1000 and 2000 assets so I think it will scale well.

Reply:
So on average 10ms additional overhead, correct?

Reply:
The median is the line in these box plots, around 3ms.